### PR TITLE
feat(cmd): support incrbyfloat

### PR DESCRIPTION
**Pull request by happy-v587**
## Summary by CodeRabbit

* **New Features**
  * Added INCRBYFLOAT command to atomically increment/decrement values by floating-point amounts; results are formatted as integers when applicable.
  * Command rejects invalid numeric inputs, NaN, and infinite increments with clear errors.

* **Tests**
  * Added comprehensive tests covering type validation, invalid inputs, edge cases, precision, and standard float operations.

### CODE REVIEW

File: src/storage/src/redis_strings.rs [hunk 1]
```diff
@@ -454,4 +454,77 @@ impl Redis {
...
+            let mut string_value = StringValue::new(format!("{}", value).to_owned());
+            string_value.set_ctime(ctime);
+            string_value.set_etime(etime);
+            let cf = self
+                .get_cf_handle(ColumnFamilyIndex::MetaCF)
+                .context(OptionNoneSnafu {
+                    message: "cf is not initialized".to_string(),
+                })?;
+            let mut batch = rocksdb::WriteBatch::default();
+            batch.put_cf(&cf, string_key.encode()?, string_value.encode());
```
Comment (guozhihao-224) [inline src/storage/src/redis_strings.rs:523]: 复用第一次的 key 的encode 值，避免重复编码

File: src/storage/src/redis_strings.rs [hunk 2]
```diff
@@ -454,4 +454,77 @@ impl Redis {
...
+        if value.is_nan() || value.is_infinite() {
+            return Err(RedisErr {
+                message: "increment would produce NaN or Infinity".to_string(),
+                location: Default::default(),
+            });
+        }
+
+        // set new value
+        {
+            let mut string_value = StringValue::new(format!("{}", value).to_owned());
```
Comment (guozhihao-224) [inline src/storage/src/redis_strings.rs:514]: 构造写入值时直接 format!("{}", value).into_bytes()，避免多余 to_owned， .to_owned() 会创建第二个 String，再次分配 & 拷贝内容。

File: src/storage/src/redis_strings.rs [hunk 3]
```diff
@@ -454,4 +454,77 @@ impl Redis {
...
+        // Get lock for the key
+        let key_str = String::from_utf8_lossy(key).to_string();
+        let _lock = ScopeRecordLock::new(self.lock_mgr.as_ref(), &key_str);
+
+        // get value by key
+        let string_key = BaseKey::new(key);
+        let encode_value = db
+            .get_opt(&string_key.encode()?, &self.read_options)
+            .context(RocksSnafu)?
+            .unwrap_or_else(Vec::new);
```
Comment (guozhihao-224) [inline src/storage/src/redis_strings.rs:472]: 使用 unwrap_or_default() 代替 unwrap_or_else(Vec::new).

File: src/storage/src/redis_strings.rs [hunk 4]
```diff
@@ -454,4 +454,77 @@ impl Redis {
...
+
+        // check key type
+        self.check_type(encode_value.as_slice(), DataType::String)?;
```
Comment (guozhihao-224) [inline src/storage/src/redis_strings.rs:475]: 只在值未过期时做类型校验，避免因已过期的错误类型触发报错

File: src/storage/src/redis_strings.rs [hunk 5]
```diff
@@ -454,4 +454,77 @@ impl Redis {
...
+        // convert user_value to f64
+        if !encode_value.is_empty() {
+            let decode_value = ParsedStringsValue::new(&encode_value[..])?;
+            // check ttl
+            if !decode_value.is_stale() {
+                let user_value = decode_value.user_value();
+                value = match String::from_utf8_lossy(&user_value).to_string().parse() {
+                    Ok(v) => v,
+                    Err(_) => {
+                        return Err(RedisErr {
```
Comment (guozhihao-224) [inline src/storage/src/redis_strings.rs:490]: 应该用RedisErrSnafu { message: ... }.fail() 或在闭包中用 .build()

**Review by AlexStocks — APPROVED** [review]
